### PR TITLE
docs: GitHub Readme improvements

### PR DESCRIPTION
**Pull request by KevinMusgrave**
## Description

The purpose of this PR is to make the GitHub Readme more interesting, and make Determined more compelling.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,75 +1,103 @@
...
+with det.core.init() as core_context:
+    ...
+```
+
+## Command Line Interface (CLI)
 
-# Navigate to a Determined example.
-git clone --recurse-submodules https://github.com/determined-ai/determined
-cd determined/examples/computer_vision/cifar10_pytorch
+Use the CLI to start the Determined cluster locally, or on your favorite cloud service:
```
Comment (stoksc) [inline README.md:41]: I imagine local deployment is the most common starter, so should it get priority over `det deploy aws` in the listing?

File: README.md [hunk 2]
```diff
@@ -1,75 +1,103 @@
-# Determined: Deep Learning Training Platform
-
 <p align="center"><img src="determined-logo.png" alt="Determined AI Logo"></p>
 
-Determined is an open-source deep learning training platform that makes building
-models fast and easy. Determined enables you to:
+Determined is an all-in-one deep learning platform, compatible with PyTorch and Tensorflow.
```
Comment (tara-det-ai) [inline README.md:3]: Determined is an all-in-one deep learning platform, compatible with PyTorch and TensorFlow.

**Review by tara-det-ai — COMMENTED** [review]
Determined is an all-in-one deep learning platform, compatible with PyTorch and TensorFlow.

**Review by sirredbeard — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,91 +1,130 @@
...
-  hyperparameters
+It takes care of:
 
-Determined integrates these features into an easy-to-use, high-performance deep
-learning environment — which means you can spend your time building models
-instead of managing infrastructure.
+- **Distributed training**, for faster results.
+- **Hyperparameter tuning**, for obtaining the best models.
+- **Resource management**, for cutting cloud GPU costs.
+- **Experiment tracking**, for analysis and reproducibility.
```
Comment (tara-det-ai) [inline README.md:10]: - Distributed training for faster results.
- Hyperparameter tuning for obtaining the best models.
- Resource management for cutting cloud GPU costs.
- Experiment tracking for analysis and reproducibility.

File: README.md [hunk 4]
```diff
@@ -1,91 +1,130 @@
...
+```python
+import determined as det
+
+with det.core.init() as core_context:
+    ...
+```
+
+## Command Line Interface (CLI)
+
+Use the CLI to start the Determined cluster locally:
```
Comment (tara-det-ai) [inline README.md:41]: You can use the CLI to:

File: README.md [hunk 5]
```diff
@@ -1,91 +1,130 @@
...
 
-# Navigate to a Determined example.
-git clone --recurse-submodules https://github.com/determined-ai/determined
-cd determined/examples/computer_vision/cifar10_pytorch
+```
+det deploy local cluster-up
+```
```
Comment (tara-det-ai) [inline README.md:45]: - Start a Determined cluster locally:

```
det deploy local cluster-up
```

File: README.md [hunk 6]
```diff
@@ -1,91 +1,130 @@
...
 
-# Submit job to train a single model on a single node.
-det experiment create const.yaml .
- ```
+Or on cloud services like AWS and GCP:
 
-## Detailed Installation Guide
+```
+det deploy aws up
+```
```
Comment (tara-det-ai) [inline README.md:51]: - Launch Determined on cloud services, such as Amazon Web Services (AWS) or Google Cloud Platform (GCP):

```
det deploy aws up
```

File: README.md [hunk 7]
```diff
@@ -1,91 +1,130 @@
...
 
-See [our installation guide](https://docs.determined.ai/latest/how-to/install-main.html) for details on how to install Determined, including on AWS and GCP.
 
-### Try Now on AWS
+Then train your models:
+```bash
+det experiment create gpt.yaml .
+```
```
Comment (tara-det-ai) [inline README.md:57]: - Train your models:

```bash
det experiment create gpt.yaml .
```

File: README.md [hunk 8]
```diff
@@ -1,91 +1,130 @@
...
+  learning_rate:
+    type: double
+    minval: .0001
+    maxval: 1.0
+searcher:
+  name: adaptive_asha
+  metric: validation_loss
+  smaller_is_better: true
+```
+
```
Comment (tara-det-ai) [inline README.md:75]: - Configure everything from distributed training to hyperparameter tuning using YAML files:

```yaml
resources:
  slots_per_trial: 8
  priority: 1
hyperparameters:
  learning_rate:
    type: double
    minval: .0001
    maxval: 1.0
searcher:
  name: adaptive_asha
  metric: validation_loss
  smaller_is_better: true
```

File: README.md [hunk 9]
```diff
@@ -1,91 +1,130 @@
...
+![Web UI](docs/assets/readme_images/webui.png)
+
+
+# Installation
+
+Install the CLI:
+```bash
+pip install determined
+```
```
Comment (tara-det-ai) [inline README.md:89]: To install the CLI:

```bash
pip install determined
```

File: README.md [hunk 10]
```diff
@@ -1,91 +1,130 @@
...
 
-## Next Steps
+Then use `det deploy` to start the Determined cluster locally, or on cloud services like AWS and GCP.
```
Comment (tara-det-ai) [inline README.md:91]: To start a Determined cluster locally, or on cloud, use `det deploy`.

File: README.md [hunk 11]
```diff
@@ -1,91 +1,130 @@
...
 
-For a brief introduction to using Determined, check out our
-[Quick Start Guide](https://docs.determined.ai/latest/getting-started.html).
+See the following guides for all installation details:
```
Comment (tara-det-ai) [inline README.md:93]: For installation details, visit the the cluster deployment guide for your environment:

**Review by tara-det-ai — APPROVED** [review]
Added suggested edits.

**Comment by sirredbeard** [comment]
@KevinMusgrave Can you please open a new PR and incorporate the suggested edits from Tara?

File: README.md (hunk 8 above)
Comment (KevinMusgrave) [reply README.md:75]: I'll leave this un-bulleted since it's not something the CLI does.

File: README.md (hunk 10 above)
Comment (KevinMusgrave) [reply README.md:91]: I like the "then" version, because it's clear that you do it *after* the pip installation.
